components/CommentForm.js

```diff
@@ -45,41 +64,95 @@ export default function CommentForm({ taskId, onAddComment }) {
     } else {
       setIsValidMessage(true);
     }
-    const response = await toast.promise(
-      fetch("/api/comments", {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({ commentData, taskId }),
-      }),
-      {
-        pending: "Comment addition is pending",
-        success: "Comment added successfully",
-        error: "Comment not added",
+    if (commentToEdit) {
+      if (data.message.trim() === commentToEdit.message) {
+        alert("No changes were made to the form.");
+        return;
+      }
+
+      const commentData = {
+        ...commentToEdit,
+        message: data.message.trim(),
+        updatedDate: new Date(),
+      };
+
+      const response = await toast.promise(
+        fetch("/api/comments", {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(commentData),
+        }),
+        {
+          pending: "Comment updating is pending",
+          success: "Comment updated successfully",
+          error: "Comment not updated",
+        }
+      );
+      if (response.ok) {
+        onChangeCommentToEdit(null);
+        onUpdateComment();
+      }
+    } else {
+      const commentData = { message: data.message.trim(), date: new Date() };
+
+      const response = await toast.promise(
+        fetch("/api/comments", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ commentData, taskId }),
+        }),
+        {
+          pending: "Comment addition is pending",
+          success: "Comment added successfully",
+          error: "Comment not added",
+        }
+      );
+      if (response.ok) {
+        onUpdateComment();
+        event.target.reset();
+        event.target.message.focus();
       }
-    );
-    if (response.ok) {
-      onAddComment();
-      event.target.reset();
-      event.target.message.focus();
     }
   }
 
+  function handleChangeMessage() {
+    setIsValidMessage(true);
+  }
+
   return (
-    <StyledForm onSubmit={handleSubmit}>
+    <StyledForm $edit={commentToEdit} onSubmit={handleSubmit}>
       <StyledLabel htmlFor="message">
-        <StyledSpan $left={true}>*</StyledSpan>Your message:
-        {!isValidMessage && <StyledSpan>Please enter your message!</StyledSpan>}
+        {!commentToEdit && (
+          <>
+            <StyledSpan $left={true}>*</StyledSpan>Your message:
+          </>
```

Comment:
```suggestion
          <p>
            <StyledSpan $left={true}>*</StyledSpan>Your message:
          </p>
```
You should add an element here, because you are rendering text :)